packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_StateCommitmentChain.sol

```diff
@@ -53,6 +53,7 @@ contract OVM_StateCommitmentChain is iOVM_StateCommitmentChain, Lib_AddressResol
     {
         FRAUD_PROOF_WINDOW = _fraudProofWindow;
```

Comment:
Current method for setting `lastSequencerTimestamp` is:

```
batches().setGlobalMetadata(_makeBatchExtraData(0, block.timestamp));
```

Reply:
tried this:
```
batches().setGlobalMetadata(_makeBatchExtraData(0, uint40(block.timestamp)));
```

but it results in this error:
```
OVM_StateCommitmentChain
       "before each" hook for "should return the current block timestamp after deployment":
     Error: Transaction reverted: function call to a non-contract account
 ```

Reply:
The reason for this is because there's currently a circular dependency between the SCC and it's ChainStorage container:

1. The ChainStorage container's `setGlobalMetadata` function is protected by the `onlyOwner`
2. That `onlyOwner` modifier calls to the AddressManager contract to see who its owner is
3. The owner should be the SCC, but the AddressManager can't know the SCC's address until it's been deployed. 
4. The SCC can't be deployed if it is trying to call a contract that doesn't yet exist. 

The AddressManager will be replaced by Chugsplash, but if we want a nearer term fix, we could move the call to the ChainStorage container from the constructor to an `initialize()` function.

Reply:
aah thanks. I take it that's a common test error and it typically means a circular dependency? 

I'm game to add an initializer function as long as it doesn't become tech debt. Otherwise we could probably just wait for chugsplash. wdyt @smartcontracts ?

Reply:
> aah thanks. I take it that's a common test error and it typically means a circular dependency?
Not sure, it's common, but yes, if you're getting that error during a constructor call it's probably the case.